assets/section-main-product.css

```diff
@@ -1325,14 +1325,46 @@ a.product__text {
 /* Product-thumbnail snippet */
 
 .product-media-container {
+  position: relative;
   width: 100%;
   max-width: calc(100% - calc(var(--media-border-width) * 2));
 }
 
+.product-media-container.constrain-height {
+  /* amount to subtract from viewport height when calculating media height */
+  --viewport-offset: 400px;
+  /* smallest height the image is allowed to scale down to */
+  --contrained-min-height: 300px;
+  /* the larger of either the calculated viewport height or the fixed minimum */
+  --constrained-height: max(var(--contrained-min-height), calc(100vh - var(--viewport-offset)));
+  margin-right: auto;
+  margin-left: auto;
+}
+
 @media screen and (min-width: 750px) {
   .product-media-container {
     max-width: 100%;
   }
+
+  .product-media-container.constrain-height {
+    --viewport-offset: 170px;
+    --contrained-min-height: 500px;
+  }
+}
+
+.product-media-container.constrain-height.media-fit-contain {
```

Comment:
I intend for `media-fit-contain/cover` and `constrain-height` to become the common naming conventions for any section to offer either of these features, so cutting out `.product-media-container` would probably be shortsighted even if it's not required presently.

I know you didn't have the full picture with the media fit setting when you first looked, but for context, these other classes and the specific combinations of them are deliberate and shouldn't be removed for specificity. I did think about using `:where()` for maintaining minimal specificity in these cases, but decided against it because browser support is still relatively recent, just barely being within our requirements.